java-api-core/src/com/kinvey/java/query/AbstractQuery.java

```diff
@@ -102,6 +102,10 @@ public String getQueryFilterJson(JsonFactory factory) {
             jsonResult = writer.toString();
         } catch (Exception ex) {}
 
+        if (jsonResult.equals("{}")) {
```

Comment:
Since there is a case above where we are checking against `{}`, do we need to set `jsonResult` to `null` here?